src/identity/manager.rs

```diff
@@ -584,6 +584,8 @@ impl SecretManager {
     }
 
     pub async fn forget_certificate(&self, id: &Identity) {
+        // TODO: consider keeping the cert around for a minute or so to avoid churn
```

Comment:
Not sure if this is realistic: What if we churn [delete, re-add]  around the time a fetch_certificate call is made? Isn't there a (very small chance) the cert isn't there when we look it up? 

https://github.com/istio/ztunnel/blob/3f3a1af786c4fedca1fd4afab34a4a4999baa83f/src/proxy/inbound.rs#L638

1. fetch workload
2. it's cert is cleaned up
3. we fail to get the cert

A slight delay here would eliminate that rare case.